person.rb

```diff
@@ -0,0 +1,33 @@
+require './nameable'
+
+class Person < Nameable
+  attr_accessor :name, :age, :id
+  attr_reader :rentals
+
+  def initialize(age, name = 'Unknown', parent_permission: true)
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @rentals = []
+    super()
+  end
+
+  def can_use_services?
+    @parent_permission || of_age?(@age)
+  end
+
+  def correct_name
+    @name
+  end
+
+  def add_rental(rental)
+    @rentals << rental unless @rentals.include?(rental)
+  end
```

Comment:
- [ ]  (Optional) Good job implementing this method as requested however I would advise you to implement as follows as this is more direct. Your approach assumes that rentals are created elsewhere and focuses on managing a collection of existing Rental objects.

e.g

```ruby
def add_rental(book, date)
    Rental.new(date, book, self)
  end
```